### PR TITLE
Remove tokio to reduce the audit surface.

### PR DESCRIPTION
**Pull request by afranchuk**
It was minimally used.

### CODE REVIEW

**Comment by Gankra** [comment]
Does this really change much given minidump-stackwalk uses tokio? Also is really no one in the mozilla cargo-vet trust web using tokio?

**Comment by afranchuk** [comment]
Yes! We're using `minidump-unwind` on its own (for the minidump analyzer in crash reporting). Nothing delivered from mozilla central uses tokio, so otherwise there would need to be an audit of tokio to ensure it's `safe-to-deploy`.

We _could_ actually do the audit, and we have discussed possibly trusting it ourselves. No imports in the cargo-vet registry have an audit for tokio yet. It is preferable to remove it as this is the only use in the entire dependency tree. Auditing it will be a big task; it has 500+ unsafe blocks.

**Comment by Gankra** [comment]
Yeah I agree it's a ton of work, just a bit surprised it's managed to evade auditing this long. Makes sense, assuming the perf still behaves reasonably (I would assume so, code is generally bottlenecked on xul.dll's symbols sync downloading, async'ing just lets the other stuff get fetched at the same time).

**Comment by afranchuk** [comment]
I expect there's a lot of `tokio` exemptions :cry: 

The new code here should be almost the same as the old code as far as perf goes. I could have written it using atomics and whatnot (actually I have an impl lying around somewhere I could copy in) but I figured it's not a high priority.

File: breakpad-symbols/src/lib.rs [hunk 1]
```diff
@@ -569,6 +571,32 @@ fn module_key(module: &(dyn Module + Sync)) -> ModuleKey {
...
+    pub async fn get<'a, F, Fut>(&self, f: F) -> Arc<Result<T, E>>
+    where
+        F: FnOnce() -> Fut + 'a,
+        Fut: std::future::Future<Output = Result<T, E>> + 'a,
+    {
+        let mut guard = self.inner.lock().await;
+        if guard.is_none() {
+            *guard = Some(Arc::new(f().await));
+        }
+        unsafe { guard.as_ref().unwrap_unchecked() }.clone()
```
Comment (Gankra) [inline breakpad-symbols/src/lib.rs:596]: there's no way this unsafe is justified, just use normal unwrap. the compiler will probably optimize it out anyway, and it won't be a fun landmine for a refactor.

**Review by Gankra — APPROVED** [review]

**Review by gabrielesvelto — APPROVED** [review]
I'm OK with this w/ the change @Gankra requested

File: breakpad-symbols/src/lib.rs (hunk 1 above)
Comment (afranchuk) [reply breakpad-symbols/src/lib.rs:596]: Whoops, definitely not justified.

**Pull request merged**